packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -182,11 +184,66 @@ function EditableBlockBindingsPanelItems( {
 export const BlockBindingsPanel = ( { name: blockName, metadata } ) => {
 	const registry = useRegistry();
 	const blockContext = useContext( BlockContext );
-	const { bindings } = metadata || {};
 	const { removeAllBlockBindings } = useBlockBindingsUtils();
 	const bindableAttributes = getBindableAttributes( blockName );
 	const dropdownMenuProps = useToolsPanelDropdownMenuProps();
 
+	// `useSelect` is used purposely here to ensure `getFieldsList`
+	// is updated whenever there are updates in block context.
+	// `source.getFieldsList` may also call a selector via `registry.select`.
+	const _fieldsList = {};
```

Comment:
What is the reason to keep that `const` outside of `useSelect` and modify it with the selector?

Reply:
If I move the `const` inside the `useSelect`, it interprets it is a new object each time it goes through it, resulting in this warning:

```
The 'useSelect' hook returns different values when called with the same state and parameters. This can lead to unnecessary rerenders.
```

I must say I am not fully familiar with how this should work, but it is the only way I found to avoid that message.

Reply:
Can you open PR without this local variable that hides the underlying issue? I would recommend asking for help from some folks who have experienced with the internals of `useSelect` like @jsnajdr and @Mamaduka.

Reply:
Ideally you should memoize the `fieldsList` value but here it's almost impossible because the data come from so many different sources. Any `getFieldsList` callback can select from any store it wants.

The easiest solution is to:
- return the `fieldsList` as the top-level object in the `useSelect` result. `const fieldsList = useSelect( ... )`
- move the `canUpdateBlockBindings` select to its own `useSelect` call. The two selects don't have anything in common anyway, the select from completely different stores.

That should remove the warning.

Reply:
I've started exploring this [here](https://github.com/WordPress/gutenberg/pull/65604/files#r1773098363). As commented there, it seems just making these changes is not enough. We can keep the discussion in the other PR.